integration/test_auth.py

```diff
@@ -288,6 +288,36 @@ def test_update_user(new_user):
     assert user.custom_claims is None
     assert len(user.provider_data) == 2
 
+    user = auth.update_user(
+        new_user.uid,
+        link_provider=auth.UserProvider(
+            uid='test', provider_id='google.com', email='test@example.com',
+            display_name='Test Name', photo_url='https://test.com/user.png'))
+    assert user.uid == new_user.uid
+    assert len(user.provider_data) == 3
+
+    user = auth.update_user(
+        new_user.uid,
+            phone_number=auth.DELETE_ATTRIBUTE,
+            delete_provider_ids=['google.com'])
+    assert user.uid == new_user.uid
+    assert user.phone_number is None
+    assert len(user.provider_data) == 1
+
+    user = auth.update_user(
+        new_user.uid,
+        phone_number=phone)
+    assert user.uid == new_user.uid
+    assert user.phone_number == phone
+    assert len(user.provider_data) == 2
+
+    user = auth.update_user(
+        new_user.uid,
+            delete_provider_ids=['phone', 'google.com'])
+    assert user.uid == new_user.uid
+    assert user.phone_number is None
+    assert len(user.provider_data) == 1
+
```

Comment:
A new requirement is that this API should also work with non-federated providers. (It's not something we're going to actually recommend, since there's more direct ways to do that.) Unfortunately, the backend for this API doesn't support non-federated providers, so we're going to have to fake that a bit.

Concretely, you've used update_user() to link to the 'google.com' provider. Could you also add calls for phone, email, oauth and saml? (Not necessarily in this exact test.) It'll require you to also change the implementation.

If this looks like it'll cause this PR to balloon in size, then let's talk first and see if we can come up with other options.